src/mlx/warnings/warnings.py

```diff
@@ -406,8 +408,13 @@ def warnings_logfile(warnings, log):
     for file_wildcard in log:
         if glob.glob(file_wildcard):
             for logfile in glob.glob(file_wildcard):
-                with open(logfile, 'r') as loghandle:
-                    warnings.check(loghandle.read())
+                _filename, file_extension = os.path.splitext(logfile)
+                if file_extension == ".cvs" or file_extension == ".tsv":
+                    with open(logfile, newline="") as file:
+                        warnings.check(file, file_extension=file_extension)
+                else:
+                    with open(logfile, 'r') as loghandle:
+                        warnings.check(loghandle.read(), file_extension=file_extension)
```

Comment:
Then I need to check if the type is `_io.TextIOWrapper` because it will be still a string when using command or in test are sometimes strings given to the function

Reply:
Resolved with adding `check_logfile` function in 68c7aa22b53182efe96b0e1d7971cbf5fb1416ba